libnavigation-core/src/main/java/com/mapbox/navigation/core/replay/history/ReplayHistoryMapper.kt

```diff
@@ -129,6 +132,16 @@ class ReplayHistoryMapper private constructor(
          * @return [ReplayHistoryMapper]
          */
         fun build(): ReplayHistoryMapper {
+            if (setRouteMapper === DefaultSetRouteMapper) {
+                LoggerProvider.logger.w(
+                    Tag("ReplayHistoryMapper"),
+                    Message(
+                        "Mapper uses a deprecated default `setRouteMapper` based on " +
+                            "`ReplaySetRoute` type. Consider switching to a mapper based on " +
+                            "`ReplayNavigationSetRoute`."
+                    )
+                )
+            }
```

Comment:
`ReplayHistoryEventMapper` is a generic mapper. Why not make the `DefaultSetRouteMapper` detect if the set route event can be converted to the new NavigationRoute?

Reply:
This warning will alarm even when using a mapper that correctly handles NavigationRoute, so doesn't seem right.